### PR TITLE
Feature/gallery favourite

### PR DESCRIPTION
**Pull request by kailasnadh790**
## Ticket/Issue number(s)
Issue #3 

## Test URLs
- Before: https://main--firefly--adobecom.hlx.page/drafts/kailas/kailas-index#
- After: https://feature-gallery-favourite--firefly--adobecom.hlx.page/drafts/kailas/kailas-index#

Fixes:

- Randomizing Gallery images on every load of the page.
- implemented images overflowing beyond the viewmore button like on the live site.
Screenshot 2024-05-16 at 11 09 20 AM

- ESlint fixes
- Gallery style fixes

### CODE REVIEW

File: blocks/firefly-gallery/firefly-gallery.js [hunk 1]
```diff
@@ -108,21 +113,51 @@ async function addCards(cardContainer, images) {
...
+      card.addEventListener('click', () => {
+        window.open(communityUrl, '_self');
+      });
+
+      favorite.addEventListener('click', async (e) => {
+        e.preventDefault();
+        if (window.adobeIMS.isSignedInUser()) {
+          const accessToken = window.adobeIMS.getAccessToken();
+          const imageId = image.urn.split(':').pop();
+          const url = FAVOURITE_URL.replace('$', imageId);
```
Comment (kailasnadh790) [inline blocks/firefly-gallery/firefly-gallery.js:126]: This is TBD.

File: blocks/firefly-gallery/firefly-gallery.js [hunk 2]
```diff
@@ -108,21 +113,51 @@ async function addCards(cardContainer, images) {
...
+          const headers = new Headers({
+            'X-Api-Key': 'alfred-community-hubs',
+            'community_id': 'ff_community',
```
Comment (kailasnadh790) [inline blocks/firefly-gallery/firefly-gallery.js:129]: Favourite button feature is still to be completed.

File: blocks/firefly-gallery/firefly-gallery.js [hunk 3]
```diff
@@ -108,21 +113,51 @@ async function addCards(cardContainer, images) {
...
+      });
     }
   });
+  // set height of container
+  cardContainer.style.height = `${heightOfContainer}px`;
   GETTING_IMAGES = false;
 }
 
 export default async function decorate(block) {
-  let size = SHORT_GALLERY_SIZE;
```
Comment (amol-anand) [inline blocks/firefly-gallery/firefly-gallery.js:117]: why pull in 48 when we need only 20 for the short gallery size

**Review by amol-anand — APPROVED** [review]

File: blocks/firefly-gallery/firefly-gallery.js (hunk 3 above)
Comment (kailasnadh790) [reply blocks/firefly-gallery/firefly-gallery.js:117]: @amol-anand On the live site, the overflow effect is achieved this way. There, it renders 48 images and shows only ~ 20 images, I am trying to achieve similar effect here by setting the height of the container equal to height of first 5 images (as the container is a 4x5 table) and setting overflow: hidden for the container.

**Pull request merged**